kobo-uncaged/main.go

```diff
@@ -502,60 +479,42 @@ func (ku *KoboUncaged) saveDeviceInfo() error {
 
 func (ku *KoboUncaged) saveCoverImage(contentID string, size image.Point, imgB64 string) {
 	defer ku.wg.Done()
-	imgID := imgIDFromContentID(contentID)
-	imgDir := path.Join(ku.bkRootDir, ku.genImageDirPath(imgID))
-	_, libFullSize, libGridSize := ku.device.CoverSize()
 
-	if err := os.MkdirAll(imgDir, 0744); err != nil {
+	img, err := imaging.Decode(base64.NewDecoder(base64.StdEncoding, strings.NewReader(imgB64)))
+	if err != nil {
 		log.Println(err)
 		return
 	}
+	sz := img.Bounds().Size()
 
-	imgBin, err := base64.StdEncoding.DecodeString(imgB64)
-	if err != nil {
-		return
+	imgDir := ".kobo-images"
+	if ku.useSDCard {
+		imgDir = "koboExtStorage/images-cache"
 	}
 
-	// Now we do our resizing
-	origCover, err := imaging.Decode(bytes.NewReader(imgBin))
-	if err != nil {
-		log.Println(err)
-		return
-	}
+	imgID := imgIDFromContentID(contentID)
 
-	var libFW, libFH, gridFW, gridFH int
-	if (float64(size.X) / float64(size.Y)) < 1.0 {
-		libFH, gridFH = libFullSize.Y, libGridSize.Y
-	} else {
-		libFW, gridFW = libFullSize.X, libGridSize.X
-	}
+	for _, cover := range []koboCover{libFull, libGrid} {
+		nsz := resizeKeepAspectRatioByExpanding(sz, cover.Size(ku.device))
+		nimg := imaging.Resize(img, nsz.X, nsz.Y, imaging.Linear)
```

Comment:
Which image resize library/algorithm are we planning to use?

Reply:
We may have to upscale the full cover, so an algorithm that's only decent for downscaling probably isn't the best idea.

Rez seemed pretty quick, so I would be happy with that. Bilinear or Bicubic perhaps. Or we could make the algorithm selectable in the config file, and the quality purists among us can select a better algorithm if they want to take the performance hit.